src/OpalCompiler-Core/IRTranslator.class.st

```diff
@@ -92,7 +92,7 @@ IRTranslator >> pushOuterVectors: scope [
 	| scopesWithVector sc |
 	scopesWithVector := OrderedCollection new.
 	sc := scope.
-	[ sc outerScope isBehavior ] whileFalse: [ 
+	[ sc outerScope isNil or: [ sc outerScope isBehavior ] ] whileFalse: [ 
```

Comment:
Why is there a new isNil check now there?

Reply:
Previously the most outer scope was a class of method. Now the top is semantics scope. 
I was needed the minimal change here to be compatible with both cases to allow this change to be made: 
- during the development I does the testing/debugging with new compiler class (a copy of original one). Then I pushed the changes to the main class.

So the logic here needs to be improved once it will be integrated. And it can be based on a scope API instead of hardcoding the type of object here (something like #isTopScope method or similar).

Reply:
Previously the most outer scope was a class of method. Now the top is semantics scope.
I was needed the minimal change here to be compatible with both cases to allow this change to be made:

during the development I does the testing/debugging with new compiler class (a copy of original one). Then I pushed the changes to the main class.
So the logic here needs to be improved once it will be integrated. And it can be based on a scope API instead of hardcoding the type of object here (something like #isTopScope method or similar).